### PR TITLE
MBS-11263 composite test runner build for fast iteration testing

### PR DESCRIPTION
**Pull request by dsvoronin**
first part with only runtime dependencies

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -134,6 +134,9 @@ stage_ui_tests:
 	make publish_to_maven_local
 	./gradlew $(project) $(log_level) $(params) :android-test:ui-testing-core-app:instrumentationUi -DinfraVersion=local
 
+test_runner_sample:
```
Comment (eugene-krivobokov) [inline Makefile:137]: sample --> instrumentation
?

File: settings.gradle.kts [hunk 2]
```diff
@@ -3,3 +3,4 @@ rootProject.name = "avito-android-infra"
 includeBuild("libraries")
 includeBuild("build-logic")
 includeBuild("subprojects")
+includeBuild("samples")
```
Comment (eugene-krivobokov) [inline settings.gradle.kts:6]: In the future, it might be reasonable to put in a test runner project, as a part of it.

**Review by eugene-krivobokov — APPROVED** [review]

File: settings.gradle.kts (hunk 2 above)
Comment (dsvoronin) [reply settings.gradle.kts:6]: yep, but also as composite build

File: Makefile (hunk 1 above)
Comment (dsvoronin) [reply Makefile:137]: done

File: samples/gradle.properties [hunk 3]
```diff
@@ -0,0 +1,7 @@
+android.useAndroidX=true
+# Mandatory properties in root project for insturmentation plugin TODO remove all of them
+teamcityBuildId=1
```
Comment (sergioboi) [inline samples/gradle.properties:3]: it strange that we need this garbage there

Comment (sergioboi) [reply samples/gradle.properties:3]: I'm about teamcityBuildId teamcityUrl buildNumber teamcityBuildType

File: samples/test-runner/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+android {
+    defaultConfig {
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+
+        // TODO: describe in docs that they are updated in IDE configuration only after sync!
+        testInstrumentationRunnerArguments(
+            mapOf(
+                "planSlug" to "AndroidTestApp"
```
Comment (sergioboi) [inline samples/test-runner/build.gradle.kts:16]: Looks redundunt

File: samples/gradle.properties (hunk 3 above)
Comment (dsvoronin) [reply samples/gradle.properties:3]: it should be removed yes, feel free to do it

File: samples/test-runner/build.gradle.kts (hunk 4 above)
Comment (dsvoronin) [reply samples/test-runner/build.gradle.kts:16]: sure, some more leftovers

**Pull request merged**